test/test.js

```diff
@@ -8,6 +8,7 @@ require.config({
 });
 
 var allTests = [
```

Comment:
does it need to be var? shouldnt' we use ES6, let?

Reply:
@montoyamoraga 
yes , we should  !
in fact i have just checked and it works fine as it should be !
:grin:

Reply:
@endurance21 can you include that in this PR? i only know that unit testing exists, but i don't know what they are for, or how to check them, i am assigning this PR to somebody else for review :)

Reply:
@montoyamoraga 
however we must also consider using `const` here , as the reference is not changing ayway !

Reply:
> @endurance21 can you include that in this PR? i only know that unit testing exists, but i don't know what they are for, or how to check them, i am assigning this PR to somebody else for review :)

for sure i will be adding it !

Reply:
but i think i have to make another pr for that , otherwise there might  be merge conflicts with other PRs !  :confused:
i will make another pr for that once other things  of this PR  get reviewed and ready to go !
:grinning: